master/buildbot/steps/source/git.py

```diff
@@ -387,6 +387,8 @@ def _fetch(self, _):
             if rc == RC_SUCCESS:
                 fetch_required = False
 
+        abandonOnFailure = not self.retryFetch and not self.clobberOnFailure
```

Comment:
side note: `not (self.retryFetch or self.clobberOnFailure)` might be a bit easier to ready.  YMMV, though